src/p528/LineOfSight.cpp

```diff
@@ -0,0 +1,294 @@
+ |                terminal_2    - Struct containing high terminal parameters
+ |                f__mhz        - Frequency, in MHz
+ |                A_dML__db     - Diffraction loss at d_ML, in dB
+ |                p             - Time percentage
+ |                d__km         - Path length, in km
+ |                T_pol         - Code indicating either polarization
+ |                                  + 0 : POLARIZATION__HORIZONTAL
+ |                                  + 1 : POLARIZATION__VERTICAL
+ |
+ |      Outputs:  los_params    - Struct contianing LOS parameters
+ |                result        - Struct contianing P.528 results
+ |                K_LOS         - K-value
+ |
+ |      Returns:  [void]
+ |
+ *===========================================================================*/
+void LineOfSight(Path *path, Terminal *terminal_1, Terminal *terminal_2, LineOfSightParams *los_params, 
+    double f__mhz, double A_dML__db, double p, double d__km, int T_pol, Result *result, double *K_LOS)
+{
+    double psi;
+    double R_Tg;
+
+	// 0.2997925 = speed of light, gigameters per sec
+    double lambda__km = 0.2997925 / f__mhz;                             // [Eqn 6-1]
+	double terminate = lambda__km / 1e6;
+
+	// determine psi_limit, where you switch from free space to 2-ray model
+	// lambda / 2 is the start of the lobe closest to d_ML
+	double psi_limit = FindPsiAtDeltaR(lambda__km / 2, path, terminal_1, terminal_2, terminate);
+
+	// "[d_y6__km] is the largest distance at which a free-space value is obtained in a two-ray model
+	//   of reflection from a smooth earth with a reflection coefficient of -1" [ES-83-3, page 44]
+	double d_y6__km = FindDistanceAtDeltaR(lambda__km / 6, path, terminal_1, terminal_2, terminate);
+
+    /////////////////////////////////////////////
+    // Determine d_0__km distance
+    //
+
+	// In IF-73, the values for d_0 (d_d in IF-77) were found to be too small when both antennas are low,
+	// so this "heuristic" was developed to fix that
+	// [Eqns 8-2 and 8-3]
+	if (terminal_1->d_r__km >= path->d_d__km || path->d_d__km >= path->d_ML__km)
+	{
+		if (terminal_1->d_r__km > d_y6__km || d_y6__km > path->d_ML__km)
+			path->d_0__km = terminal_1->d_r__km;
+		else
+			path->d_0__km = d_y6__km;
+	}
+	else if (path->d_d__km < d_y6__km && d_y6__km < path->d_ML__km)
+		path->d_0__km = d_y6__km;
+	else
+		path->d_0__km = path->d_d__km;
+
+    //
+    // Determine d_0__km distance
+    /////////////////////////////////////////////
+
+    /////////////////////////////////////////////
+    // Tune d_0__km distance
+    //
+
+    // Now that we have d_0, lets carefully walk it forward, 1 meter at a time, to tune it to as precise as possible without
+    //      going beyond the LOS region (ie, beyond d_ML)
+    double d_temp__km = path->d_0__km;
+    while (true)
+    {
+        psi = FindPsiAtDistance(d_temp__km, path, terminal_1, terminal_2);
+
+        LineOfSightParams result;
```

Comment:
The var name `result` is already used (a param). Might be good to rename this one.